spec/requests/media/odk_response_spec.rb

```diff
@@ -25,6 +25,20 @@
       expect(form_response.answers.count).to eq(2)
       expect(form_response.odk_xml_file_name).to eq("submission.xml")
       expect(form_response.odk_xml_file_size).to be > 0
+
+      tmp_files = Dir.glob(ResponsesController::TMP_UPLOADS_PATH.join("*.xml"))
+      expect(FileUtils.rm(tmp_files)).to be_empty
+    end
+
+    it "should save a temp file for failures" do
+      submission_file = prepare_and_upload_submission_file("no_version.xml")
+
+      post submission_path, params: {xml_submission_file: submission_file},
+                            headers: {"HTTP_AUTHORIZATION" => encode_credentials(user.login, test_password)}
+      expect(response).not_to have_http_status(:created)
+
+      tmp_files = Dir.glob(ResponsesController::TMP_UPLOADS_PATH.join("*.xml"))
+      expect(FileUtils.rm(tmp_files)).not_to be_empty
```

Comment:
You should probably clean out the `TMP_UPLOADS_PATH` dir before each of these specs (in a before block? use `FileUtils.rm_rf`?). Else you will end up with cross contamination between runs potentially.